perf/src/main/scala/cromwell/perf/Call.scala

```diff
@@ -0,0 +1,69 @@
+package cromwell.perf
+
+import java.time.{Duration, OffsetDateTime}
+
+import io.circe.JsonObject
+
+case class CallCaching(hit: Option[Boolean],
+                       result: Option[String],
+                       hitFailures: Option[Seq[Map[String, Seq[JsonObject]]]])
+
+
+case class ExecutionEvents(startTime: OffsetDateTime,
+                           description: String,
+                           endTime: OffsetDateTime)
+
+
+case class Call(shardIndex: Int,
+                start: OffsetDateTime,
+                end: OffsetDateTime,
+                callCaching: Option[CallCaching],
+                executionEvents: Seq[ExecutionEvents]) {
+  val callCachingEventStates = List("CheckingCallCache", "FetchingCachedOutputsFromDatabase", "BackendIsCopyingCachedOutputs")
+  val jobPreparationEventStates = List("Pending", "RequestingExecutionToken", "WaitingForValueStore", "PreparingJob", "CheckingJobStore")
+  val cacheCopyingEventStates = List("FetchingCachedOutputsFromDatabase", "BackendIsCopyingCachedOutputs")
+
+  /**
+    * @return Cache copy retries before getting a successful hit or running the job
+    */
+  def cacheCopyRetries: Int = {
+    val hitFailuresMap = callCaching.map(callCachingObject => callCachingObject.hitFailures.getOrElse(Map.empty))
+
+    if (hitFailuresMap.isDefined) hitFailuresMap.get.size else 0
```

Comment:
TOL

```
    val numFailures: Option[Int] = for {
      caching <- callCaching
      failures <- caching.hitFailures
    } yield failures.size
    
    numFailures.getOrElse(0)
```

Reply:
This looks more cleaner! Will change it. Thanks :)